app/jenkins-for-jira-ui/src/components/JenkinsServerList/JenkinsServerList.tsx

```diff
@@ -23,19 +23,38 @@ const JenkinsServerList = (): JSX.Element => {
 	const analyticsClient = new AnalyticsClient();
 	const [jenkinsServers, setJenkinsServers] = useState<JenkinsServer[]>();
 	const [moduleKey, setModuleKey] = useState<string>();
+	const isMountedRef = React.useRef<boolean>(true);
+
 	const fetchAllJenkinsServers = async () => {
-		const servers = await getAllJenkinsServers() || [];
-		setJenkinsServers(servers);
+		try {
+			const servers = await getAllJenkinsServers() || [];
```

Comment:
/nit `getAllJenkinsServers` looks like would return [] anyway so [] not required.
You could probably move `const servers = await getAllJenkinsServers()` inside that con as well / return early before it